metagraph/src/graph/representation/succinct/dbg_succinct.hpp

```diff
@@ -14,6 +16,15 @@ class DBGSuccinct : public DeBruijnGraph {
   public:
     friend class MaskedDeBruijnGraph;
 
+    /**
+     * A wrapper which caches computed node sequences for DBGSuccinct graphs.
+     * This allows for faster get_node_sequence and call_incoming_kmers calls.
+     * Traversal operations and map_to_nodes_sequentially are overridden to fill
+     * the cache alongside their normal functions.
+     * Bytes stored per k-mer: ~56 (k<=21), ~64 (k<=42), ~80 (k<=85)
+     */
+    class CachedView;
```

Comment:
Where should this class be used?

I know you need it for alignment, but anything else?

This description doesn't make it clear how and in which cases this helps.
If there is a need to do some caching, where is the guarantee that this class will do a good job? Maybe there will be another caching strategy more useful for the application.

In general, I'm thinking about the expediency of this think if this indeed should be a general thing on this level rather than one of the alignment modules.

Reply:
The caching strategy was designed to speed up these particular methods since they are the slowest of the frequently-used methods.

Any kind of operation which requires a sequence of backwards traversals can benefit from this. I don't think it's a good idea to make this an alignment module since it's a graph traversal helper and not something particular to alignment.

Reply:
So what does it cache exactly, on what operation? Add this to the description.

Reply:
I updated the comment to add more details.